### PR TITLE
Fixes #16273 - allow execmem

### PR DESCRIPTION
**Pull request by lzap**
FFI library callback feature executes mapped memory of temporary files. This
sounds dangerous, however FFI authors do not think this is a problem, they did
not find a workaround in the FFI code as a good option and they recommend to
allow this in SELinux in the official documentation:

* https://bitbucket.org/cffi/cffi/issues/231
* https://cffi.readthedocs.io/en/latest/using.html#callbacks

This patch enables execmem both for passenger and smart proxy process as there
is no other good solution to this problem.

### CODE REVIEW

**Comment by ekohl** [comment]
This is only needed by Zest which is only used in Katello. Perhaps a candidate for katello-selinux?

**Comment by lzap** [comment]
Do you insist? I'd rather be precausious and since we've run into this in the past (we actually had execmem for some time in core) I am for having this in core.

**Comment by ekohl** [comment]
You typically want to allow as little as possible and execmem sounds scary. If core doesn't use it, it'd be good to avoid it which was my reason to move it to katello-selinux.

**Comment by ehelms** [comment]
I'd love to see katello-selinux merged here so there is only a single selinux policy across the ecosystem if that helps.

**Comment by ekohl** [comment]
I have considered a katello selinux boolean in this package so we have one package, but only allow the minimum needed.

**Comment by lzap** [comment]
I've split this into foreman-proxy part (here) and katello-selinux (passenger): https://github.com/Katello/katello-selinux/pull/19

**Comment by lzap** [comment]
This is ready for merge, @tbrisker wanted us to unblock nightly. After this is merged I will go ahead and push updated packages.

**Review by ekohl — APPROVED** [review]
Not happy this is needed, but I guess we don't really have a choice here.

**Review by tbrisker — APPROVED** [review]

**Comment by lzap** [comment]
Nightly with the patch is out: http://yum.theforeman.org/nightly/el7/x86_64/foreman-selinux-1.22.0-0.1.develop.20190403150122git390a568.el7.noarch.rpm

Thanks.

**Comment by tbrisker** [comment]
@lzap iirc you said we should add an upgrade warning regarding this to the manual, care to open a PR against nightly with it?

**Pull request merged**
